src/Bicep.Core/TypeSystem/Providers/ThirdParty/ThirdPartyResourceTypeLoader.cs

```diff
@@ -0,0 +1,56 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using Azure.Bicep.Types;
+using Azure.Bicep.Types.K8s;
+using Bicep.Core.Resources;
+using Bicep.Core.TypeSystem.Providers.Az;
+using Bicep.Core.TypeSystem.Providers.K8s;
+using Bicep.Core.TypeSystem.Types;
+using static System.Runtime.InteropServices.JavaScript.JSType;
+
+namespace Bicep.Core.TypeSystem.Providers.ThirdParty
+{
+    public class ThirdPartyResourceTypeLoader : IResourceTypeLoader
+    {
+        private readonly ITypeLoader typeLoader;
+        //No need to create a 3rd party factory
+        private readonly K8sResourceTypeFactory resourceTypeFactory;
```

Comment:
There shouldn't be a reference to `K8s` here - do you know why this is necessary?

Reply:
Renamed to ExtensibilityResourceTypeFactory in the following commit